### ISSUE TITLE
Leave Federation

### ISSUE THREAD

**Issue by justinmoon** [fedimint/fedimint#3272]
Conceptually clients can "leave a federation". AFAIK fedimint doesn't have any code to implement this anywhere, but I'm sure many of us have deleted client data directories created by `fedimint-cli` once they are no longer needed.

We should formalize what exactly it means to "leave a federation". In particular, we may want to add a method to `ClientModule` to cleanup any pending contracts which could cause the users to lose funds if they "left" the federation immediately.

For example:
* Cancel lightning offers so nobody pays the corresponding invoice funding an account which wouldn't be accessible once recipient leaves the federation.
* Sweep any lightning sends stuck in the refund case, but this requires waiting.
* External modules like the "stability pool" will need to withdraw from price-lock contracts into ecash.

**Comment by justinmoon** [comment]
Quick fix idea: just check for active operations and don't allow "leaving" if there are any

**Comment by tvolk131** [comment]
For my own understanding, your quick fix idea would involve:

1. Checking that no client funds are in-flight into/out of a federation
2. Checking that no client funds are at rest in a federation
3. Deleting client-side federation data

Would we also want to tell the federation that it can wipe any backup data for the given user?

**Comment by justinmoon** [comment]
@tvolk131 Yes. We could say that (3) could be responsibility of whatever developer is integrating the client. (2) is pretty trivial. (3) is what needs work. Our trait system for the client needs a way to report "no client funds are in-flight into/out of a federation".

**Comment by tvolk131** [comment]
> (3) is what needs work.

Typo? I think you mean (1).

**Comment by justinmoon** [comment]
> Typo? I think you mean (1).

Yes. Fixed.
